design/pushed-authorization-requests.md

```diff
@@ -0,0 +1,326 @@
+
+These parameters should be added to the .well-known/openid-configuration:
+
+- **pushed_authorization_request_endpoint**
+
+The URL of the pushed authorization request endpoint at which a client can post an authorization request in exchange for a "request_uri".
+
+The presence of "pushed_authorization_request_endpoint" is sufficient for a client to determine that it may use the pushed authorization requests flow. A "request_uri" value obtained from the PAR endpoint is usable at the authorization endpoint regardless of other authorization server metadata such as "request_uri_parameter_supported" or "require_request_uri_registration"
+
+- **require_pushed_authorization_requests**
+
+Boolean parameter indicating whether the authorization server accepts authorization request data only via the pushed authorization request method. If omitted, the default value is "false"
+
+
+### Client Metadata
+
+The following client metadata parameter indicate whether pushed authorization requests are required for the given client
+
+- **require_pushed_authorization_requests**
+
+Boolean parameter indicating whether the only means of initiating an authorization request the client is allowed to use is a pushed authorization request. If omitted, the default value is "false"
+
+
+
+### PAR endpoint URL
+
+It should be found out from the pushed_authorization_request_endpoint in the Authorization server metadata and has this form:
+
+````
+{scheme}://{host}[:{port}]/par
+````
+
+
+### API overview
+
+The pushed authorization request endpoint is an HTTP API at the authorization server that accepts HTTP "POST" requests with parameters in the HTTP request entity-body using the "application/x- www-form-urlencoded" format with a character encoding of UTF-8.
+
+##### Header parameters:
+
+-   **[ Authorization ]** Used for HTTP basic authentication of the client, if applicable.
+
+-   **Content-Type** Must be set to *`application/x-www-form-urlencoded`*
+
+##### Body:
+
+-   The OAuth 2.0 authorisation request parameters.
+
+##### Success:
+
+-   Code: `201`
+
+-   Content-Type: `application/json`
+
+-   Body: {object} The PAR Response.
+
+##### Errors:
+
+-   400 (Bad Request) with a PAR Error 
+
+The authorization server returns an error response with the same format as is specified for error responses from the Authorization endpoint and MUST refuse any other request with HTTP status code 400 and error code "invalid_request".
+
+In addition to the above, the pushed authorization request endpoint can also make use of the following HTTP status codes:
+
+-   405 (Method Not Allowed)
+
+If the request did not use the "POST" method, the authorisation server responds with an HTTP 405 (Method Not Allowed) status code.
+
+-   413 (Payload Too Large)
+
+If the request size was beyond the upper bound that the authorization server allows, the authorization server responds with an HTTP 413 (Payload Too Large) status code
+
+-   429 (Too Many Requests)
```

Comment:
If supporting this error, keycloak needs to keep how frequent a client sends requests on PAR endpoint. It might be good that Implementation details sections describe it.

Reply:
Since it's not mandatory according to specification and adds some complexity, we will not support it in the first version.